docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,392 @@
+.. note::
+
+    More information about using administrator roles (ACL/RBAC) can be found :doc:`here </book/customers/admin_user>`.
+
+* Create ``App\Checker\ResourceChannelEnabilibityChecker`` and decorate ``Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface``.
+
+* Then add ``Supplier`` as checking resource:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Checker;
+
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface;
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityChecker as DecoratedResourceChannelEnabilibityChecker;
+
+    final class ResourceChannelEnabilibityChecker implements ResourceChannelEnabilibityCheckerInterface
+    {
+        /** @var DecoratedResourceChannelEnabilibityChecker */
+        private $decoratedResourceChannelEnabilibityChecker;
+
+        public function __construct(DecoratedResourceChannelEnabilibityChecker $decoratedResourceChannelEnabilibityChecker)
+        {
+            $this->decoratedResourceChannelEnabilibityChecker = $decoratedResourceChannelEnabilibityChecker;
+        }
+
+        public function forResourceName(string $resourceName): bool
+        {
+            if (!$this->decoratedResourceChannelEnabilibityChecker->forResourceName($resourceName)) {
+                return in_array($resourceName, [
+                    'supplier'
+                ]);
+            }
+
+            return true;
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Checker\ResourceChannelEnabilibityChecker:
+        decorates: Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface
+        arguments: ['@.inner']
+
+* Create ``App\Checker\ResourceChannelChecker`` and decorate ``Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface`` next add condition for checking ``Supplier``.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Checker;
+
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface;
+    use Sylius\Plus\Entity\ChannelInterface;
+
+    final class ResourceChannelChecker implements ResourceChannelCheckerInterface
+    {
+        /** @var ResourceChannelCheckerInterface */
+        private $decoratedResourceChannelChecker;
+
+        public function __construct(ResourceChannelCheckerInterface $decoratedResourceChannelChecker)
+        {
+            $this->decoratedResourceChannelChecker = $decoratedResourceChannelChecker;
+        }
+
+        public function isFromChannel(object $resource, ChannelInterface $channel): bool
+        {
+                // this condition is needed because while creating a Symfony form (our edit/create section), the given object has all fields as null. After creating an empty form this iteration is repeated and properly checked.            if ($resource instanceof Supplier && $resource->getChannel() === null) {
```

Comment:
```suggestion
            // this condition is needed because while creating a Symfony form (our edit/create section), the given object has all fields as null. After creating an empty form this iteration is repeated and properly checked.            
            if ($resource instanceof Supplier && $resource->getChannel() === $channel) {
```